dataset/src/test/scala/frameless/functions/NonAggregateFunctionsTests.scala

```diff
@@ -17,6 +16,71 @@ class NonAggregateFunctionsTests extends TypedDatasetSuite {
     super.afterAll()
   }
 
+  test("ceil big decimal") {
+    val spark = session
+    import spark.implicits._
+
+    def prop[A: TypedEncoder: Encoder]
+    (values: List[X1[A]])
+    (
+      implicit catalystRound: CatalystRound[A, java.math.BigDecimal],
+      encX1:Encoder[X1[A]]
+    )= {
+      val cDS = session.createDataset(values)
+      val resCompare = cDS
+        .select(sparkFunctions.ceil(cDS("a")))
+        .map(_.getAs[java.math.BigDecimal](0))
+        .collect()
+        .toList.map(_.setScale(0))
+
+      val typedDS = TypedDataset.create(values)
+      val col = typedDS('a)
+      val res = typedDS
+        .select(ceil(col))
+        .collect()
+        .run()
+        .toList
+
+      resCompare ?= res
+    }
+
+    check(forAll(prop[BigDecimal] _))
+  }
+
+  test("ceil") {
+    val spark = session
+    import spark.implicits._
+
+    def prop[A: TypedEncoder : Encoder]
+    (values: List[X1[A]])
+    (
+      implicit catalystAbsolute: CatalystRound[A, Long],
```

Comment:
Does something prevent you from turning this `Long` into a type parameter of the prop? That would allow you to factor out the `Long` and the `java.math.BigDecimal` tests into a single prop.

Reply:
ok, joined into single prop